### PR TITLE
funding: move Manager to own package

### PR DESCRIPTION
**Pull request by Crypt-iQ**
Renames `fundingManager` to `Manager`. Moves the `Manager` to the `funding` package. Will help a lot with unit testing. The last commit should be viewed with a git client with `--color-moved` as GitHub is too picky to see the `fundingmanager.go -> funding/manager.go` as a code-move.

### CODE REVIEW

File: fmgr/log.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package fmgr
+
+import (
+	"github.com/btcsuite/btclog"
+	"github.com/lightningnetwork/lnd/build"
+)
+
+// Subsystem defines the logging code for this subsystem.
+const Subsystem = "FMGR"
```
Comment (halseth) [inline fmgr/log.go:9]: also register logger in this commit?

File: fundingmanager_test.go [hunk 2]
```diff
@@ -657,25 +657,26 @@ func fundChannel(t *testing.T, alice, bob *testNode, localFundingAmt,
 
 	// Create a funding request and start the workflow.
 	errChan := make(chan error, 1)
-	initReq := &openChanReq{
```
Comment (halseth) [inline fundingmanager_test.go:659]: can delete the definition of `openChanReq`?

File: fmgr/manager.go [hunk 3]
```diff
@@ -1,12 +1,38 @@
 package fmgr
 
 import (
+	"bytes"
```
Comment (halseth) [inline fmgr/manager.go:4]: vert good idea using `--color-moved` 👍

**Review by halseth — APPROVED** [review]
LGTM

File: fundingmanager.go [hunk 4]
```diff
@@ -120,8 +120,25 @@ func (r *reservationWithCtx) updateTimestamp() {
 // embedded within this message giving the funding manager full context w.r.t
 // the workflow.
 type InitFundingMsg struct {
-	peer lnpeer.Peer
-	*openChanReq
+	Peer             lnpeer.Peer
+	TargetPubkey     *btcec.PublicKey
+	ChainHash        chainhash.Hash
```
Comment (bhandras) [inline fundingmanager.go:184]: nit: could add some godocs too.

**Review by bhandras — APPROVED** [review]
LGTM :100:

q: would it be more idiomatic Go to have the package name as `funding` and then rename `FundingManager` as `Manager`?

File: fundingmanager_test.go (hunk 2 above)
Comment (Crypt-iQ) [reply fundingmanager_test.go:659]: Yes good idea

File: fundingmanager.go (hunk 4 above)
Comment (Crypt-iQ) [reply fundingmanager.go:184]: This will replace `openChanReq` so going to refactor a little and add godocs.

**Comment by Crypt-iQ** [comment]
> q: would it be more idiomatic Go to have the package name as `funding` and then rename `FundingManager` as `Manager`?

I think `funding` is a better name for the package and I agree with the rename to `Manager`.

**Comment by Crypt-iQ** [comment]
If the package is named `funding` and what was previously the `fundingManager` is now the `Manager`, then what will the interface that `Manager` satisfies be called?

**Comment by Crypt-iQ** [comment]
The interface is named `Controller` - also removed the `PendingChannels()` from the `Manager` as it was totally unused. Perhaps this visibility can be added later once the code is cleaned up a bit.

File: funding/interfaces.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package fmgr
+package funding
```
Comment (halseth) [inline funding/interfaces.go:1]: nice, I like this 👍

File: fundingmanager.go [hunk 6]
```diff
@@ -196,15 +196,15 @@ type fundingConfig struct {
...
+	// SendAnnouncement is used by the Manager to send announcement
 	// messages to the Gossiper to possibly broadcast to the greater
 	// network. A set of optional message fields can be provided to populate
 	// any information within the graph that is not included in the gossip
 	// message.
 	SendAnnouncement func(msg lnwire.Message,
 		optionalFields ...discovery.OptionalMsgField) chan error
 
-	// NotifyWhenOnline allows the FundingManager to register with a
+	// NotifyWhenOnline allows the Manager to register with a
```
Comment (halseth) [inline fundingmanager.go:334]: I don't really think we need to change these comments. talking about the `FundingManager` is okay even though it is named `Manager` now

File: fundingmanager.go [hunk 7]
```diff
@@ -444,24 +444,24 @@ func newFundingManager(cfg fundingConfig) (*fundingManager, error) {
 
 // Start launches all helper goroutines required for handling requests sent
 // to the funding manager.
-func (f *fundingManager) Start() error {
+func (m *Manager) Start() error {
```
Comment (halseth) [inline fundingmanager.go:574]: I would prefer keeping these `f` actually 🙈

File: funding/log.go [hunk 8]
```diff
@@ -0,0 +1,27 @@
+package funding
+
+import (
+	"github.com/btcsuite/btclog"
+	"github.com/lightningnetwork/lnd/build"
+)
+
+// Subsystem defines the logging code for this subsystem.
+const Subsystem = "FMGR"
```
Comment (halseth) [inline funding/log.go:9]: re-use `FNDG`? Also can remove the old `fndgLog`

File: log.go [hunk 9]
```diff
@@ -82,7 +82,6 @@ var (
 	ltndLog = addLndPkgLogger("LTND")
 	rpcsLog = addLndPkgLogger("RPCS")
 	srvrLog = addLndPkgLogger("SRVR")
-	fndgLog = addLndPkgLogger("FNDG")
```
Comment (halseth) [inline log.go:84]: ok, removed here, nice. Still think it is better to keep the log prefix as before though.

**Review by halseth — CHANGES_REQUESTED** [review]

File: fundingmanager.go (hunk 6 above)
Comment (Crypt-iQ) [reply fundingmanager.go:334]: yeah

**Review by bhandras — APPROVED** [review]
LGTM, nice refactors! @Crypt-iQ

**Review by halseth — APPROVED** [review]
Nice, I like this latest iteration a lot! Easy to follow commits, LGTM 👍

**Review by bhandras — APPROVED** [review]
LGTM :100:

**Pull request merged**